packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/RealmMap.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.types
+
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.internal.RealmMapEntrySet
+import io.realm.kotlin.internal.RealmMapMutableEntry
+import io.realm.kotlin.notifications.DictionaryChange
+import io.realm.kotlin.notifications.InitialDictionary
+import io.realm.kotlin.notifications.UpdatedDictionary
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * TODO
+ */
+public interface RealmMap<K, V> : MutableMap<K, V>
+
+/**
+ * TODO
+ */
+public interface RealmDictionary<E> : RealmMap<String, E> {
```

Comment:
Nitpick, but should we use `<String, V>` instead of `<String, E>`? So it matches the RealmMap names.